### ISSUE TITLE
Indicate which props triggered a change

### ISSUE THREAD

**Issue by OliverJAsh** [maicki/why-did-you-update#12]
Also see issue from version prior to this fork: https://github.com/garbles/why-did-you-update/issues/10

**Comment by OliverJAsh** [comment]
I have some WIP for this. What do you think to the patch below? If you like this direction I can raise a PR.

``` patch
diff --git a/node_modules/why-did-you-update/lib/deepDiff.js b/node_modules/why-did-you-update/lib/deepDiff.js
index 747af17..77060a0 100644
--- a/node_modules/why-did-you-update/lib/deepDiff.js
+++ b/node_modules/why-did-you-update/lib/deepDiff.js
@@ -43,6 +43,8 @@ var DIFF_TYPES = {
 
 exports.DIFF_TYPES = DIFF_TYPES;
 var classifyDiff = function classifyDiff(prev, next, name) {
+  var keys = (0, _lodashUnion2['default'])((0, _lodashKeys2['default'])(prev), (0, _lodashKeys2['default'])(next));
+
   if (prev === next) {
     return {
       type: DIFF_TYPES.SAME,
@@ -52,12 +54,17 @@ var classifyDiff = function classifyDiff(prev, next, name) {
     };
   }
 
+  var isNotSame = function isNotSame(key) {
+    return prev[key] !== next[key];
+  };
+  var notSameKeys = (0, _lodashFilter2['default'])(keys, isNotSame);
+
   if ((0, _lodashIsEqual2['default'])(prev, next)) {
     return {
       type: DIFF_TYPES.EQUAL,
       name: name,
-      prev: prev,
-      next: next
+      prev: (0, _lodashPick2['default'])(prev, notSameKeys),
+      next: (0, _lodashPick2['default'])(next, notSameKeys)
     };
   }
 
@@ -70,24 +77,23 @@ var classifyDiff = function classifyDiff(prev, next, name) {
     };
   }
 
-  var isChanged = function isChanged(key) {
-    return prev[key] !== next[key] && !(0, _lodashIsEqual2['default'])(prev[key], next[key]);
-  };
   var isSameFunction = function isSameFunction(key) {
     var prevFn = prev[key];
     var nextFn = next[key];
     return (0, _lodashIsFunction2['default'])(prevFn) && (0, _lodashIsFunction2['default'])(nextFn) && prevFn.name === nextFn.name;
   };
 
-  var keys = (0, _lodashUnion2['default'])((0, _lodashKeys2['default'])(prev), (0, _lodashKeys2['default'])(next));
-  var changedKeys = (0, _lodashFilter2['default'])(keys, isChanged);
+  var isUnequal = function isUnequal(key) {
+    return isNotSame(key) && !(0, _lodashIsEqual2['default'])(prev[key], next[key]);
+  };
+  var unequalKeys = (0, _lodashFilter2['default'])(keys, isUnequal);
 
-  if (changedKeys.length && (0, _lodashEvery2['default'])(changedKeys, isSameFunction)) {
+  if (unequalKeys.length && (0, _lodashEvery2['default'])(unequalKeys, isSameFunction)) {
     return {
       type: DIFF_TYPES.FUNCTIONS,
       name: name,
-      prev: (0, _lodashPick2['default'])(prev, changedKeys),
-      next: (0, _lodashPick2['default'])(next, changedKeys)
+      prev: (0, _lodashPick2['default'])(prev, unequalKeys),
+      next: (0, _lodashPick2['default'])(next, unequalKeys)
     };
   }
 

```

**Comment by aaron-harvey** [comment]
Is this similar to PR #9 ?

**Comment by OliverJAsh** [comment]
It is the same, my apologies. Do you still want me to raise the above patch as a PR or should I leave it?

**Comment by OliverJAsh** [comment]
Closed by #9
